src/plugins/intel_gna/src/optimizer/gna_pass_manager.cpp

```diff
@@ -62,6 +62,76 @@ std::shared_ptr<IPassManager> BasePass::getPassManager() {
     return sharedMgr;
 }
 
+static auto DoNotSkip = [](CNNLayerPtr) {
+    return false;
+};
+
+/**
+ * @brief Return true if the layer has max one non-1 dimension
+ * (because we can treat it as a single dimension layer, then)
+ */
+static bool IsOneDimLayer(CNNLayerPtr layer) {
+    auto dims = layer->insData[0].lock()->getDims();
+    return std::count_if(std::begin(dims), std::end(dims), [](size_t dim) { return dim > 1; }) <= 1;
```

Comment:
We already have something similar:
is_one_dim_shape
Take a look as well at gna_graph_patterns.hpp.

Reply:
is_one_dim_shape() is not applicable to CNN layer. 
Moved IsOneDimLayer to gna_graph_patterns.hpp.